src/foam/mlang/mlangJava.js

```diff
@@ -399,24 +399,22 @@ foam.CLASS({
   methods: [
     {
       name: 'compare',
+      javaReturns: 'int',
+      args: [
+        {
+          name: 'o1',
+          javaType: 'Object'
+        },
+        {
+          name: 'o2',
+          javaType: 'Object'
+        }
+      ],
       javaCode: 'return -1 * getArg1().compare(o1, o2);'
     },
     {
       name: 'toString',
       javaCode: 'return "DESC(" + getArg1().toString() + ")";'
-    },
-    {
-      name: 'orderTail',
```

Comment:
You should remove the following functions from the `foam.mlang.order.Comparator` interface as well. Otherwise you will have issues with this class not implementing all the functions of that interface.